src/js/components/auditlogs/eventdetails/terminalsession.js

```diff
@@ -0,0 +1,107 @@
+import React, { Fragment, useEffect, useState } from 'react';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import Time from 'react-time';
+import moment from 'moment';
+import momentDurationFormatSetup from 'moment-duration-format';
+
+import { Launch as LaunchIcon } from '@material-ui/icons';
+
+import { getDeviceById, getSessionDetails } from '../../../actions/deviceActions';
+import { getIdAttribute } from '../../../selectors';
+import theme from '../../../themes/mender-theme';
+import Loader from '../../common/loader';
+import { Terminal } from '../../devices/terminal';
+import { formatTime } from '../../../helpers';
+
+momentDurationFormatSetup(moment);
+
+const BEGINNING_OF_TIME = '2020-01-01T00:00:00.000Z';
+
+export const TerminalSession = ({ device, idAttribute, item, getDeviceById, getSessionDetails }) => {
+  const [sessionDetails, setSessionDetails] = useState();
+
+  useEffect(() => {
+    const { action, actor, meta, object, time } = item;
+    if (!device) {
+      getDeviceById(object.id);
+    }
+    getSessionDetails(
+      meta.session_id[0],
+      object.id,
+      actor.id,
+      action.startsWith('open') ? time : undefined,
+      action.startsWith('close') ? time : undefined
+    ).then(setSessionDetails);
+  }, []);
+
+  if (!(sessionDetails && device)) {
+    return <Loader show={true} />;
+  }
+
+  const { name, device_type, artifact_name } = device.attributes;
+  const usesId = !idAttribute || idAttribute === 'id' || idAttribute === 'Device ID';
```

Comment:
`'Device ID'` is used a bit all over the code, perhaps time to define it? (not for this PR, but maybe one day)

Reply:
yep... definitely planned... it gets all the more important with the vision of relying also on custom names in the future...